base/range.jl

```diff
@@ -342,12 +342,12 @@ done(r::UnitRange, i) = i==oftype(i,r.stop)+1
 
 ## indexing
 
-getindex(r::Range, i::Real) = getindex(r, to_index(i))
-
-function getindex{T}(r::Range{T}, i::Integer)
+function getindex(r::Range, i::Integer)
     1 <= i <= length(r) || throw(BoundsError())
-    convert(T, first(r) + (i-1)*step(r))
+    unsafe_getindex(r, i)
 end
+unsafe_getindex{T}(v::Range{T}, i::Integer) = convert(T, first(v) + (i-1)*step(v))
```

Comment:
I've been looking forward to being able to skip bounds checks on Ranges for a long time. Yay!